src/deepsparse/pipeline_loggers/prometheus_pipeline_logger.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any
+
+from deepsparse import PipelineLogger
+from prometheus_client import Histogram, start_http_server
+
+
+__all__ = ["PrometheusLogger"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PrometheusLogger(PipelineLogger):
+    """
+    PipelineLogger that uses the official Prometheus python client
+    (https://github.com/prometheus/client_python) to monitor the
+    inference pipeline
+
+    :param port: the port used by the client. Default is 8000
+    """
+
+    def __init__(self, port: int = 8000):
+        self.port = port
+        self.setup_client()
+        # the data structure responsible for the instrumentation
+        # of the metrics
+        self.metrics = []
+
+        super().__init__()
+
+    def setup_client(self):
+        """
+        Starts the Prometheus client
+        """
+        start_http_server(port=self.port)
+        _LOGGER.info(f"Prometheus client: started. Using port: {self.port}.")
+
+    def log_latency(
+        self, inference_timing: "InferenceTimingSchema", **kwargs  # noqa F821
+    ):
+        """
+        Continuously logs the inference pipeline latencies
+
+        :param inference_timing: Pydantic model that contains information
+        about time deltas processes within the inference pipeline
+        """
+        if not self.metrics:
+            # will be run on the first digestion of
+            # inference_timing data
+            self._setup_metrics(inference_timing)
+
+        self._assert_incoming_data_consistency(inference_timing)
+
+        for metric_name, value_to_log in dict(inference_timing).items():
+            self._log_latency(metric_name=metric_name, value_to_log=value_to_log)
+
+        _LOGGER.info("Prometheus client: logged latency metrics")
```

Comment:
definitely don't want to log to console every time we log to prometheus, especially with no extra info